app/models/sushi.rb

```diff
@@ -1,7 +1,33 @@
 # frozen_string_literal:true
 
 module Sushi
+  mattr_accessor :info
+
   class << self
+    ##
+    # @param dates [Array<String>]
+    # @raise [Sushi::InvalidParameterValue] when either of the given dates are not formatted correctly.
+    #
+    # @return [TrueClass]
+    #
+    # @note This is necessary because calling `to_date` on a string in 'MM-YYYY' format will assume that 'MM' is the day,
+    #       and will set the current month as 'MM' instead.
+    #
+    #       params.fetch(:begin_date) => "06-2023"
+    #       params.fetch(:begin_date).to_date => Wed, 06 Sep 2023
+    def validate_date_format(dates = [])
+      dates.each do |date|
+        match = date.match(/(^\d{4}-\d{2}$)|(^\d{4}-\d{2}-\d{2}$)/)
```

Comment:
```
'2023-06-10'.match(/(^\d{4}-\d{2}$)|(^\d{4}-\d{2}-\d{2}$)/) => #<MatchData "2023-06-10" 1:nil 2:"2023-06-10">
'2023-06'.match(/(^\d{4}-\d{2}$)|(^\d{4}-\d{2}-\d{2}$)/) => #<MatchData "2023-06" 1:"2023-06" 2:nil>
'06-2023'.match(/(^\d{4}-\d{2}$)|(^\d{4}-\d{2}-\d{2}$)/) => nil
```

Reply:
Can you write tests for these?  Regular expressions are notorious for missing use cases.

Reply:
@jeremyf other than the two specs in sushi_spec.rb?

Reply:
Ugh, I completely mentally skipped over those.